assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -1442,60 +1482,41 @@ subroutine clamp_variable(dom_id, var_index, variable)
 
 ! if we get here, either the min, max or both have a clamping value.
   
-!>@todo this is what the code needs to be for CLM and any other
-! model that allows missing values in the state.  right now that
-! is defined in assim_tools_mod but i don't think we can use it
-! because of circular module dependencies.  it should be defined
-! maybe in filter?  and set into some low level module (like types
-! or constants or options_mod so anyone can query it).
-!
-! if we allow missing values in the state (which jeff has never
-! liked because it makes the statistics funny), then these next
-! two lines need to be:
 allow_missing = get_missing_ok_status()
 
 if (allow_missing) then
-   my_minmax(1) = minval(variable, mask=(variable /= missing_r8))
-   my_minmax(2) = maxval(variable, mask=(variable /= missing_r8))
+   ! by the 'native' missing value ("replace_missing()").
+   call get_missing_value(dom_id, var_index, mymissing)
+   my_minmax(1) = minval(variable, mask=(variable /= mymissing))
+   my_minmax(2) = maxval(variable, mask=(variable /= mymissing))
```

Comment:
Each variable may have its own missing value.  Since we've already checked that if it has both a `_FillValue` and `missing_value` that they are equal, we can replace the DART missing with the single known value.